### PR TITLE
[test]Add string testcase to search

### PR DESCRIPTION
**Pull request by jingkl**
Signed-off-by: jingkl <jingjing.jia@zilliz.com>
issue #16833

### CODE REVIEW

File: tests/python_client/testcases/test_search.py [hunk 1]
```diff
@@ -0,0 +1,2989 @@
...
+        method: create collection and insert data 
+                create index and collection load
+                collection search uses string expr in string field, string field is not primary
+        expected: Search successfully
+        """
+        # 1. initialize with data
+        collection_w, _, _, insert_ids = \
+            self.init_collection_general(prefix, True, auto_id=auto_id, dim=default_dim)[0:4]
+        # 2. search
+        log.info("test_search_normal_default_params: searching collection %s" % collection_w.name)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2743]: Here is wrong.

Comment (binbinlv) [reply tests/python_client/testcases/test_search.py:2743]: test_search_normal_default_params -> test_search_normal_default_params

File: tests/python_client/testcases/test_search.py [hunk 2]
```diff
@@ -0,0 +1,2989 @@
...
+        method: create collection and insert data 
+                create index and collection load
+                collection search uses string expr in string field ,string field is primary
+        expected: Search successfully
+        """
+        # 1. initialize with data
+        collection_w, _, _, insert_ids = \
+            self.init_collection_general(prefix, True, dim=dim, primary_field=ct.default_string_field_name)[0:4]
+        # 2. search
+        log.info("test_search_normal_default_params: searching collection %s" % collection_w.name)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2772]: test_search_normal_default_params-> test_search_normal_default_params

File: tests/python_client/testcases/test_search.py [hunk 3]
```diff
@@ -0,0 +1,2989 @@
...
+        method: create collection and insert data
+                create index and collection load
+                collection search uses mix expr 
+        expected: Search successfully
+        """
+        # 1. initialize with data
+        collection_w, _, _, insert_ids = \
+            self.init_collection_general(prefix, True, auto_id=auto_id, dim=dim)[0:4]
+        # 2. search
+        log.info("test_search_normal_default_params: searching collection %s" % collection_w.name)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2801]: test_search_normal_default_params-> test_search_normal_default_params

File: tests/python_client/testcases/test_search.py [hunk 4]
```diff
@@ -0,0 +1,2989 @@
...
+                            output_fields=output_fields,
+                             _async=_async,
+                            travel_timestamp=0,
+                            check_task=CheckTasks.check_search_results,
+                            check_items={"nq": default_nq,
+                                         "ids": insert_ids,
+                                         "limit": default_limit,
+                                         "_async": _async})
+
+    @pytest.mark.tags(CaseLabel.L0)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2816]: I think this could be L2

File: tests/python_client/testcases/test_search.py [hunk 5]
```diff
@@ -0,0 +1,2989 @@
...
+            if not expression or eval(expression):
+                filter_ids.append(_id)
+
+        # 2. create index
+        index_param = {"index_type": "IVF_FLAT", "metric_type": "L2", "params": {"nlist": 100}}
+        collection_w.create_index("float_vector", index_param)
+        collection_w.load()
+
+        # 3. search with expression
+        log.info("test_search_with_expression: searching with expression: %s" % expression)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2872]: Same issue

File: tests/python_client/testcases/test_search.py [hunk 6]
```diff
@@ -0,0 +1,2989 @@
...
+                            default_limit, default_search_string_exp,
+                            _async=_async,
+                            check_task=CheckTasks.check_search_results,
+                            check_items={"nq": default_nq,
+                                         "ids": insert_ids,
+                                         "limit": 2,
+                                         "_async": _async})
+
+
+    @pytest.mark.tags(CaseLabel.L0)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2958]: This could be L2

File: tests/python_client/testcases/test_search.py [hunk 7]
```diff
@@ -0,0 +1,2989 @@
...
+        if _async:
+            search_res.done()
+            search_res = search_res.result()
+
+        filter_ids_set = set(filter_ids)
+        for hits in search_res:
+            ids = hits.ids
+            assert set(ids).issubset(filter_ids_set)
+
+    @pytest.mark.tags(CaseLabel.L0)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2891]: This could be L1

File: tests/python_client/testcases/test_search.py [hunk 8]
```diff
@@ -0,0 +1,2989 @@
...
+                            _async=_async,
+                            travel_timestamp=0,
+                            check_task=CheckTasks.check_search_results,
+                            check_items={"nq": default_nq,
+                                         "ids": insert_ids,
+                                         "limit": default_limit,
+                                         "_async": _async})
+        
+
+    @pytest.mark.tags(CaseLabel.L0)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_search.py:2788]: This could be L2

File: tests/python_client/testcases/test_search.py (hunk 4 above)
Comment (jingkl) [reply tests/python_client/testcases/test_search.py]: Already modified

File: tests/python_client/testcases/test_search.py (hunk 6 above)
Comment (jingkl) [reply tests/python_client/testcases/test_search.py]: Already modified

File: tests/python_client/testcases/test_search.py (hunk 7 above)
Comment (jingkl) [reply tests/python_client/testcases/test_search.py]: Already modified

File: tests/python_client/testcases/test_search.py (hunk 8 above)
Comment (jingkl) [reply tests/python_client/testcases/test_search.py]: Already modified

**Pull request merged**